middleware.go

```diff
@@ -41,13 +41,16 @@ type Config struct {
 	ClientErrorLevel slog.Level
 	ServerErrorLevel slog.Level
 
-	WithRequestID      bool
-	WithRequestBody    bool
-	WithRequestHeader  bool
-	WithResponseBody   bool
-	WithResponseHeader bool
-	WithSpanID         bool
-	WithTraceID        bool
+	WithRequestIP        bool
```

Comment:
I would return the request IP and body size every time.

(I didn't run a benchmark to measure the overhead of the dump)

The `bodyWriter` can be adapted to skip the internal buffer and just sum total bytes.

Reply:
since you use the `middleware.NewWrapResponseWriter(w, r.ProtoMajor)` we don't even need to adapt the `bodyWriter` right?

Reply:
honestly I'm pretty uncomfortable with the current way of recording the request body size.

I want to use this middleware with a file upload service, which can deal with files in the range of multiple gigabytes. Reading the whole body into memory is kinda bad here

Reply:
I went ahead & wrote a `bodyReader` based off your `bodyWriter`.
It records the read bytes and optionally writes the read bytes into it's own buffer while respecting the max size

I also went ahead & modified the `bodyWriter` to utilize the `middleware.WrapResponseWriter` since it already handles all what we want